pkg/controller/workload/cache/workload_cache.go

```diff
@@ -70,9 +70,36 @@ func composeNetworkAddress(network string, addr netip.Addr) NetworkAddress {
 	}
 }
 
-func (w *cache) AddWorkload(workload *workloadapi.Workload) {
+func (w *cache) getUniqueServicesOnLeftWorkload(workload1, workload2 *workloadapi.Workload) []string {
+	var diff []string
+	if workload1 == nil {
+		return diff
+	}
+
+	for key := range workload1.Services {
+		if workload2 == nil {
+			diff = append(diff, key)
+			continue
+		}
+		if _, exist := workload2.Services[key]; !exist {
+			diff = append(diff, key)
+		}
+	}
+	return diff
+}
+
+func (w *cache) compareWorkloadServices(workload1, workload2 *workloadapi.Workload) ([]string, []string) {
+	dels := w.getUniqueServicesOnLeftWorkload(workload1, workload2)
+	news := w.getUniqueServicesOnLeftWorkload(workload2, workload1)
```

Comment:
we can construct two sets from bothj workload, and then we can use sets.Difference

Reply:
Good suggestion. However, we need to build two mapsets and add the service names of the old and new workloads to mapset. The code is not simplified. Therefore, I'll keep the modify.